From e6a87386dec135bcee8dcca9f3140cedb71f554e Mon Sep 17 00:00:00 2001
From: mio <miyopan@e.email>
Date: Wed, 14 Oct 2026 04:54:36 +0000
Subject: [PATCH] Fix build with SWIG 4.5.0

Python 2 API macros were removed in SWIG 4.5.0. Update the python
bindings to use the Python 3 API.
---
 binding/python/team/capi.i.in | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/binding/python/team/capi.i.in b/binding/python/team/capi.i.in
index c7a5842..de1542e 100644
--- a/binding/python/team/capi.i.in
+++ b/binding/python/team/capi.i.in
@@ -21,7 +21,7 @@ void py_team_change_handler_unregister(struct team_handle *th,
 %cstring_output_allocate(char **mode_name, )
 
 %typemap(in) (char *ifname, unsigned int maxlen) {
-    $2 = PyInt_AsLong($input);
+    $2 = PyLong_AsLong($input);
     $1 = (char *) malloc(sizeof(char) * $2);
 }
 
@@ -33,7 +33,7 @@ void py_team_change_handler_unregister(struct team_handle *th,
 int team_hwaddr_set(struct team_handle *th, uint32_t ifindex, const char *addr, unsigned int addr_len);
 
 %typemap(in) (char *addr, unsigned int addr_len) {
-    $2 = PyInt_AsLong($input);
+    $2 = PyLong_AsLong($input);
     $1 = (char *) malloc(sizeof(char) * $2);
 }
 %typemap(freearg) (char *addr, unsigned int addr_len) {
@@ -111,8 +111,8 @@ static int PythonCallBack(struct team_handle *th, void *func_priv,
 
     if (result == NULL)
         return -1;
-    if (PyInt_Check(result))
-        retval = PyInt_AsLong(result);
+    if (PyLong_Check(result))
+        retval = PyLong_AsLong(result);
     if (arglist != NULL)
         Py_DECREF(arglist);
     Py_XDECREF(result);
-- 
2.56.0

